File: 백준/Silver/11501. 주식/주식.kt
import java.io.BufferedReader
import java.io.File
import java.io.InputStreamReader
import java.util.*

fun main() {
    val os = System.getProperty("os.name").lowercase(Locale.getDefault())
    val args: List<String> = if (os.contains("linux")) {
        val br = BufferedReader(InputStreamReader(System.`in`))
        br.readLines()
    } else {
        val inputFile = File("src/main/kotlin/input.txt")
        inputFile.readLines()
    }
    val n = args[0].toInt()
    val rest = args.slice(1 until args.size)

    fun solve(n: Int, stocks: List<Long>) {
        var sum: Long = 0
        var max: Long = 0
        for (i in n - 1 downTo 0) {
            max = maxOf(max, stocks[i])
            if (stocks[i] < max) {
                sum += max - stocks[i]
            }
        }
        println(sum)
    }
    for (i in 0 until n * 2 step 2) {
        solve(rest[i].toInt(), rest[i + 1].split(' ').map { it.toLong() })
    }

}

File: 백준/Silver/11501. 주식/주식.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt"
const input = fs.readFileSync(filePath)
  .toString()
  .trim()
  .split("\n")


function solution([n, ...args]) {
  n = +n
  for (let i = 0; i < n * 2; i += 2) {
    solve(+args[i], args[i + 1].split(' ').map(Number))
  }
}

const solve = (n, stocks) => {
  let sum = 0
  for(let i = n - 1; i >= 0; i--){
    let current = stocks[i]
    let count = 0
    for(let j = i - 1; j >= 0; j--){
      let target = stocks[j]
      if(target < current){
        sum += current - target
        count++
      } else break
    }
    i -= count
  }
  console.log(sum)
}

solution(input)
